raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -410,6 +411,7 @@ public void becomeLeader() throws Exception {
         time.sleep(electionTimeoutMs * 2L);
         expectAndGrantVotes(currentEpoch + 1);
         expectBeginEpoch(currentEpoch + 1);
+        client.poll();
```

Comment:
Why this change?

Reply:
I was thinking `expectBeginEpoch` should only check for beginEpoch requests and we should leave it to the test to call poll when needed

Reply:
I won't need `expectBeginEpoch` in my tests anymore, so I'll revert the change.